From b2e4f8456f13d3558e76eb79ec5ad3881a047b55 Mon Sep 17 00:00:00 2001
From: Scott Windsor <swindsor@gmail.com>
Date: Fri, 3 Oct 2008 14:11:22 -0700
Subject: [PATCH] Removing uneeded require introduced in cf281091

 This fixes a performance issue when calling Rails.env or
 Rails.env.development?
---
 railties/lib/initializer.rb       |    1 -
 railties/test/initializer_test.rb |    8 ++++++++
 2 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index 74d2daa..a363971 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -49,7 +49,6 @@ module Rails
     end
 
     def env
-      require 'active_support/string_inquirer'
       ActiveSupport::StringInquirer.new(RAILS_ENV)
     end
 
diff --git a/railties/test/initializer_test.rb b/railties/test/initializer_test.rb
index 5147eeb..ec5ab63 100644
--- a/railties/test/initializer_test.rb
+++ b/railties/test/initializer_test.rb
@@ -28,6 +28,14 @@ class Initializer_load_environment_Test < Test::Unit::TestCase
     $initialize_test_set_from_env = nil
   end
 
+  def test_rails_env
+    config = ConfigurationMock.new("#{File.dirname(__FILE__)}/fixtures/environment_with_constant.rb")
+    Rails::Initializer.run(:load_environment, config)
+    assert_equal "development", Rails.env
+    assert_equal true, Rails.env.development?
+    assert_equal false, Rails.env.production?
+  end
+
 end
 
 class Initializer_eager_loading_Test < Test::Unit::TestCase
-- 
1.5.4.3

